lib/redis/lock.rb

```diff
@@ -38,18 +38,19 @@ def lock(&block)
       try_until_timeout do
         expiration = generate_expiration
         # Use the expiration as the value of the lock.
-        break if redis.setnx(key, expiration)
+        break if redis.setnx(key, expiration) && Redis.next.setnx(key, expiration)
```

Comment:
I suppose this is ok:
- If the lock is being held by old redis, this will break from the block.
- If the lock is NOT being held by old redis, this will set it and then also set it in new redis. This assumes that NEW redis always mirrors OLD redis.
The lock _should_ never be held by NEW redis, and not OLD redis.